logstash-core/src/main/java/org/logstash/EventFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.logstash;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * A factory for events.
+ */
+@FunctionalInterface
+public interface EventFactory {
```

Comment:
This new interface _very nearly_ is implemented by the `co.elastic.logstash.api.EventFactory` that is available to java-based plugins through `co.elastic.logstash.api.Context` which is provided to the constructor. Can we converge without breaking consumers of `co.elastic.logstash.api.EventFactory`?

Reply:
@yaauie :100: thanks I completely missed the existing `EventFactory` - for sure, no reason to re-invent the wheel!
the only down-side is `org.logstash.Event.fromJson` assumes the concrete implementation being returned, thus we'll assume and cast the returned events.